src/hooks/useOfflineDetector.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useRef, useState } from 'react';
+
+const POLLING_DELAY = 5000;
```

Comment:
I think it would be better to have possibility to customize this interval via `UseOfflineDetectorParams`.